hamilton/lifecycle/default.py

```diff
@@ -376,7 +377,10 @@ def run_to_execute_node(
         if node_name not in self.cache_vars:
             return node_callable(**node_kwargs)
 
-        node_hash = graph_types.hash_source_code(node_callable, strip=True)
+        source_of_node_callable = node_callable
+        while isinstance(source_of_node_callable, partial):  # handle partials
```

Comment:
cool -- can make it into a function for clarity: `get_original_function(func: Callable)` + test, but this is good